client-js/app/elements/labrad-plot.ts

```diff
@@ -470,214 +826,212 @@ export class Plot extends polymer.Base {
-                .append('rect')
-                .datum(row)
-                .classed('data', true)
-                .attr('x', (d) => p.xScale(d[0]))
-                .attr('y', (d) => p.yScale(p.yNext[d[1]]))
-                .attr('width', (d) => p.xScale(p.xNext[d[0]]) - p.xScale(d[0]))
-                .attr('height', (d) => {
-                  return Math.abs(p.yScale(p.yNext[d[1]]) - p.yScale(d[1]));
-                })
-                .style('fill', (d) => getColor(d[p.displaySurface], zMin, zMax));
+        child.geometry.getAttribute('position').needsUpdate = true;
+
+        // If we are drawing dots, we need to scale them so they are a constant
+        // size relative to the screen regardless of zoom level.
+        if (this.drawMode2D == 'dots') {
+          this.projectGraphCoordToScreenRect(0, 0, screenRect);
+          const wScreen = screenRect.w;
+          this.projectScreenCoordToWorldCoord(wScreen, 0, vector);
+          const wWorldEnd = vector.x;
+          const wWorld = wWorldEnd - xWorldZero;
+          child.material.size = wWorld;
         }
-        break;
-
-      default:
-        // Nothing to do.
-        break;
+      }
     }
   }
 
 
-  private installMouseListeners() {
-    if (!this.svg) return;
-    switch (this.mouseMode) {
-      case 'pan':
-        this.svg.on('mousedown', null);
-        this.svg.call(this.zoom);
-        this.$.plot.style.cursor = 'auto';
-        this.$.plot.style.cursor = 'all-scroll';
-        break;
+  /**
+   * Projects a coordinate in graph coordinates (relative to the axes of the
+   * data) to screen coordinates (relative to the canvas). Additionally
+   * determines the width and height of the coordinate based on its type.
+   *
+   * Mutates `outputObject` returning the projected `x` and `y` coordinate as
+   * well as `w` and `h` for the projected width and height.
+   */
+  private projectGraphCoordToScreenRect(
+      x: number, y: number,
+      outputObject: {x: number, y: number, w: number, h: number}): void {
+    let wScreen, hScreen;
 
-      case 'zoomRect':
-        this.svg.on('.zoom', null)
-        this.svg.on('mousedown', () => this.zoomRectangle());
-        this.$.plot.style.cursor = 'auto';
-        this.$.plot.style.cursor = 'crosshair';
+    switch (this.drawMode2D) {
+      case 'dots':
+        // Points have a fixed width relative to the screen regardless of zoom.
+        wScreen = PLOT_POINT_SIZE;
+        hScreen = wScreen;
         break;
 
-      default:
-        // Nothing to do.
-        break;
-    }
-  }
-
-
-  private updatePlotStyles() {
-    if (!this.svg) return;
-    switch (this.numIndeps) {
-      case 1:
-        this.$$('rect.background').style.fill = '#ffffff';
-        this.$.modes2d.style.visibility = 'hidden';
-        this.is1D = true;
-        this.is2D = false;
+      case 'rectfill':
+        wScreen = Math.abs(this.xScale(this.dx0) - this.xScale(0));
+        hScreen = Math.abs(this.yScale(this.dy0) - this.yScale(0));
```

Comment:
I think you need to modify y here, otherwise the rectfill and vargrid modes render uniform grids in different places (you can see this in the uniform-grid sample plots where the plot shifts vertically when switching between these two modes):
```typescript
y += this.dy0
```